tests/components/esphome/test_config_flow.py

```diff
@@ -40,26 +46,27 @@ def mock_api_connection_error():
         yield mock_error
 
 
-def _setup_flow_handler(hass):
-    flow = config_flow.EsphomeFlowHandler()
-    flow.hass = hass
-    flow.context = {}
-    return flow
-
-
 async def test_user_connection_works(hass, mock_client):
     """Test we can finish a config flow."""
-    flow = _setup_flow_handler(hass)
-    result = await flow.async_step_user(user_input=None)
-    assert result["type"] == "form"
+    result = await hass.config_entries.flow.async_init(
+        "esphome", context={"source": "user"}, data=None,
+    )
 
-    mock_client.device_info.return_value = mock_coro(MockDeviceInfo(False, "test"))
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
 
-    result = await flow.async_step_user(user_input={"host": "127.0.0.1", "port": 80})
+    mock_client.device_info = AsyncMock(return_value=MockDeviceInfo(False, "test"))
 
-    assert result["type"] == "create_entry"
-    assert result["data"] == {"host": "127.0.0.1", "port": 80, "password": ""}
+    result = await hass.config_entries.flow.async_init(
+        "esphome",
+        context={"source": "user"},
+        data={CONF_HOST: "127.0.0.1", CONF_PORT: 80},
+    )
+
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["data"] == {CONF_HOST: "127.0.0.1", CONF_PORT: 80, CONF_PASSWORD: ""}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5451397)